core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java

```diff
@@ -1186,6 +1230,26 @@ public Server build() {
             virtualHosts = this.virtualHosts;
         }
 
+        // Gets the access logger for an each virtual host.
+        virtualHosts.forEach(vh -> {
+            if (vh.accessLogger() == null) {
+                final Logger logger = accessLoggerMapper.apply(vh);
+                if (logger == null) {
+                    throw new IllegalStateException(
+                            String.format("Access Logger for the virtual host (%s) is null",
+                                          vh.hostnamePattern()));
+                }
+                vh.accessLogger(logger);
+            }
+        });
+        if (defaultVirtualHost.accessLogger() == null) {
+            final Logger logger = accessLoggerMapper.apply(defaultVirtualHost);
+            if (logger == null) {
+                throw new IllegalStateException("Access Logger for default virtual host is null");
```

Comment:
How about: `accessLoggerMapper.apply() has returned null for the default virtual host.`